Support/lib/rspec/mate/switch_command.rb

```diff
@@ -140,8 +140,9 @@ def klass(relative_path, content=nil)
       def write_and_open(path)
         FileUtils.mkdir_p(File.dirname(path))
         described = described_class_for(path, base_dir)
+        is_rails = File.directory? path.sub(%r{/spec/.*$}, '/app')
```

Comment:
I agree the whole rails thing could use some refactoring. That said if I'm not mistaken checking for `app/` is already used to detect rails in other parts of the code.
This should work most of the time, possibly other frameworks use the same pattern inside `spec/` too…